uint32 car_position_x
uint32 car_position_y